src/main/java/uk/org/tombolo/ImporterSpecificationRunner.java

```diff
@@ -0,0 +1,120 @@
+package uk.org.tombolo;
+
+import org.apache.commons.lang.exception.ExceptionUtils;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import uk.org.tombolo.core.Attribute;
+import uk.org.tombolo.core.Datasource;
+import uk.org.tombolo.core.SubjectType;
+import uk.org.tombolo.importer.Importer;
+
+import java.lang.reflect.Constructor;
+import java.util.List;
+import java.util.stream.Collectors;
+
+/**
+ * Get a list of datasources for an importer
+ */
+public class ImporterSpecificationRunner extends AbstractRunner {
+    private static final Logger log = LoggerFactory.getLogger(ImporterSpecificationRunner.class);
+    private static final ImporterSpecificationRunner runner = new ImporterSpecificationRunner();
+
+    public static void main(String[] args) throws Exception {
+
+        boolean subjects = Boolean.parseBoolean(args[1]);
+        boolean attribute = Boolean.parseBoolean(args[2]);
+        boolean provider = Boolean.parseBoolean(args[3]);
+        String importerClass = args[0];
+        StringBuilder info = new StringBuilder("\n\n");
+
+        if (importerClass.equals("None")) {
+            CatalogueExportRunner cer = new CatalogueExportRunner();
+            List<Class<? extends Importer>> dcImporters = cer.getImporterClasses();
+            info.append("List of Digital Connector Importers: \n\n");
+            dcImporters.forEach(i -> info.append(i.getCanonicalName()).append("\n"));
+            log.info(info.toString());
+            return;
+        }
+
+        Importer importer = runner.getImporter(importerClass);
+        List<String> datasources = runner.getDatasourceIds(importer);
+        
+        for (String d : datasources) {
+            info.setLength(0);
+            Datasource dSource = runner.getDatasource(d, importer);
+            if (null != dSource.getDatasourceSpec()) {
+                info.append("\n\nDatasource Id: ")
+                    .append(dSource.getDatasourceSpec().getId())
+                    .append("\nImporter Class: ")
+                    .append(dSource.getDatasourceSpec().getImporterClass().getCanonicalName())
+                    .append("\nData Url: ")
+                    .append(dSource.getDatasourceSpec().getUrl()).append("\n");
+                
+                if (provider) {
+                    info.append("Provider: ")
+                        .append(runner.getProvider(importer)).append("\n");
+                }       
+
+                if (subjects) {
+                    info.append("SubjectTypes: ")
+                        .append(runner.getSubjectType(dSource).toString()).append("\n");
+
+                }
+                if (attribute) {
+                    info.append("Timed Value Attributes: ")
+                        .append(runner.getTimedValueAttributes(dSource).toString())
```

Comment:
I would suggest a newline between the attributes or something else to make it easier to navigate if many.

Reply:
new line could work but sometimes the attribute list is so long that it would make user scroll a lot. But if you think its okay, then i would do it

Reply:
I know, I tried it on a long list and I got lost though. This is why I suggested the insertion of new line. Personally, I would prefer to scroll then get blinded. I understand that if logging all the datasources of a specific importer that has many many attributes, would be a disaster. There is no win here.
A solution to this could be suggesting the user in the readme to pipe the output to a file, so they can search, see it better and so on.

Reply:
no worries, done.